### ISSUE TITLE
Implement logical operators || and &&

### ISSUE THREAD

**Issue by boeckmann** [boeckmann/asm6502#24]
The logical operators become necessary because we allow undefined expressions in IF statements. With a bit-wise operator, the following will be undefined and therefore lead to a wrong result:
```
X = ?
Y = 1
.IF X | Y
  .ECHO "perfect"
.ELSE
  .ERROR "should not happen"
.ENDIF
```
X | Y = ?. That makes sense for a bit-wise operation. But for the logical operation, that should be true. So the following operations are defined:
```
? || ? = 0
? || 0 = 0
0 || ? = 0
? || 1 = 1
1 || ? = 1
1 || 1 = 1
? && ? = 0
? && 0 = 0
0 && ? = 0
? && 1 = 0
1 && ? = 0
1 && 1 = 1
```

**Comment by boeckmann** [comment]
Implemented by acad86e.

**Comment by boeckmann** [comment]
The following semantics apply:
```
Logical Operators
? || ? = ?    ? && ? = ?   
? || 0 = ?    ? && 0 = ?
0 || ? = ?    0 && ? = ?
? || 1 = 1    ? && 1 = ?
1 || ? = 1    1 && ? = ?
0 || 1 = 1    0 && 1 = 0
1 || 0 = 1    1 && 0 = 0
1 || 1 = 1    1 && 1 = 1

Comparison operators: return ? if one operand is ?
? == ? = ?
? != ? = ?
? == 0 = ?
0 == ? = ?
? != 0 = ?
0 != ? = ?
? == 1 = ?
1 == ? = ?
? != 1 = ?
1 != ? = ?
```
